bundle/manifests/aws-load-balancer-operator.clusterserviceversion.yaml

```diff
@@ -66,9 +66,15 @@ metadata:
         }
       ]
     capabilities: Basic Install
+    features.operators.openshift.io/disconnected: "true"
+    features.operators.openshift.io/fips-compliant: "true"
```

Comment:
The fips bug [OCPBUGS-14846](https://issues.redhat.com/browse/OCPBUGS-14846) has been fixed by PR https://github.com/openshift/aws-load-balancer-operator/pull/99 but that's for latest v1.1.0.
If this PR is back ported to old release then we need to update this annotation to "false" I think.

Reply:
Got catch! Yes, I'll need to backport this PR down to 1.0. So the 1.0 PR needs to have `fips-compliant` set to `false`.

Reply:
I just want to double-check that this is correct.  

We set `CGO_ENABLED=0` in the Dockerfile: https://github.com/openshift/aws-load-balancer-operator/blob/1a2c320b27d4b5bb1a9c259db0796c462a11cc98/Dockerfile#L17

We don't set `CGO_ENABLED` in the Makefile: https://github.com/openshift/aws-load-balancer-operator/blob/1a2c320b27d4b5bb1a9c259db0796c462a11cc98/Makefile#L178

The Dockerfile is referenced here: https://github.com/openshift/release/blob/8703107e5b40d8559b2d7858350a88e3634762a6/ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml#L17

Do we use the Dockerfile or the Makefile?  Is `CGO_ENABLED` set or not, and if it is, to what value?  

My understanding based on https://github.com/openshift/external-dns-operator/pull/213#discussion_r1458679687, is that we need `CGO_ENABLED=1` (or maybe not setting `CGO_ENABLED` at all is sufficient?) in order to use OpenSSL, and using OpenSSL (from UBI) is sufficient to claim FIPS compliance using the `features.operators.openshift.io/fips-compliant: "true"` annotation.  Do I understand correctly?

Reply:
> Do we use the Dockerfile or the Makefile?

You found the right CI config, we use `Dockerfile` to build the operator image which is then [added to the OLM bundle](https://github.com/openshift/release/blob/8703107e5b40d8559b2d7858350a88e3634762a6/ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml#L25-L27) which in turn is [deployed](https://github.com/openshift/release/blob/8703107e5b40d8559b2d7858350a88e3634762a6/ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml#L61) for each e2e test.

> Is CGO_ENABLED set or not, and if it is, to what value?

It's set to `0` which means that we use a statically built operator for the e2e tests. Also the users of the GitHub version of ALBO may use `Dockerfile` and build not FIPS compliant operator binary. Then what we deliver to the customers is built on CPaaS where [we don't set](https://gitlab.cee.redhat.com/cpaas-midstream/aws-load-balancer-operator/-/blob/aws-lb-optr-1.1-rhel-8/distgit/containers/aws-load-balancer-operator/Dockerfile.in?ref_type=heads#L10) `CGO_ENABLED` so the operator is built for the dynamic linkage.

> My understanding based on https://github.com/openshift/external-dns-operator/pull/213#discussion_r1458679687, is that we need CGO_ENABLED=1 (or maybe not setting CGO_ENABLED at all is sufficient?) in order to use OpenSSL, and using OpenSSL (from UBI) is sufficient to claim FIPS compliance using the features.operators.openshift.io/fips-compliant: "true" annotation. Do I understand correctly?

Default value for `CGO_ENABLED` is `1`, so it's enough to not set it. Whether the UBI image is fine the FIPS compliance or not is not 100% clear to me. I tend to say yes as it's RHEL based. Asked [a question](https://docs.google.com/presentation/d/1o3IowxHX6BsnxGkIInaQ0lBgnn_K5Ex8jxwCYCeNsqs/edit?disco=AAABGTECK14) in FIPS workshop slides.

Let me remove `CGO_ENABLED=0` from `Dockerfile`.

Reply:
`CGO_ENABLED=0` removed from Dockerfile.

Reply:
https://github.com/openshift/aws-load-balancer-operator/pull/133 fixed the FIPS compliance for ALBO. I think we have the right to set `true` now.